managing-os/templates/fedora/upgrade-25-to-26.md

```diff
@@ -0,0 +1,241 @@
+    errors, see the next two points first.
+
+     * If `dnf` reports that you do not have enough free disk space to proceed
+       with the upgrade process, create an empty file in dom0 to use as a cache
+       and attach it to the template as a virtual disk.
+
+           [user@dom0 ~]$ truncate -s 5GB /var/tmp/template-upgrade-cache.img
+           [user@dom0 ~]$ qvm-block -A fedora-26 dom0:/var/tmp/template-upgrade-cache.img
+
+       Then reattempt the upgrade process, but this time use the virtual disk
+       as a cache.
+
+           [user@fedora-26 ~]$ sudo mkfs.ext4 /dev/xvdi
+           [user@fedora-26 ~]$ sudo mount /dev/xvdi /mnt/removable
+           [user@fedora-26 ~]$ sudo dnf clean all
+           [user@fedora-26 ~]$ sudo dnf --releasever=26 --setopt=cachedir=/mnt/removable --best --allowerasing distro-sync
+
+       If this attempt is successful, proceed to step 4.
+
+     * `dnf` may complain:
+
+           At least X MB more space needed on the / filesystem.
+
+       In this case, one option is to [resize the TemplateVM's disk
+       image][resize-disk-image] before reattempting the upgrade process. 
+       (See [Additional Information] below for other options.)
+
+ 4. Shut down the new TemplateVM (from the command-line or Qubes VM Manager).
+
+        [user@dom0 ~]$ qvm-shutdown fedora-26
+
+ 5. Remove the cache file, if you created one.
+
+        [user@dom0 ~]$ rm /var/tmp/template-upgrade-cache.img
+
+ 6. Trim the new template (see [Compacting the Upgraded Template] for details
+    and other options).
+
+        [user@dom0 ~]$ qvm-trim-template fedora-26
+
+ 7. (Recommended) Switch everything that was set to the old template to the new
+    template, e.g.:
+
+     1. Make the new template the default template:
+
+        Qubes Manager --> Global settings --> Default template
+
+     2. Base AppVMs on the new template. In Qubes Manager, for each VM that is
+        currently based on `fedora-25` that you would like to base on
+        `fedora-26`, enter its VM settings and change the Template selection:
+
+        Qubes Manager --> (Select a VM) --> VM settings --> Template
+
+     3. Base the [DispVM] template on the new template.
+
+        If you have set the new template as your default template:
+
+            [user@dom0 ~]$ qvm-create-default-dvm --default-template
+
+        Otherwise:
+
+            [user@dom0 ~]$ qvm-create-default-dvm fedora-26
+
+ 8. (Optional) Remove the old template. (Make sure to type `fedora-25`, not
+    `fedora-26`.)
+
+        [user@dom0 ~]$ sudo dnf remove qubes-template-fedora-25
+
+
+Summary: Upgrading the Minimal Fedora 25 Template to Fedora 26
+--------------------------------------------------------------
+
+**Note:** The prompt on each line indicates where each command should be entered
+(`@dom0` or `@fedora-26`).
+
+        [user@dom0 ~]$ qvm-clone fedora-25-minimal fedora-26-minimal
+        [user@dom0 ~]$ qvm-run -a fedora-26-minimal xterm
```

Comment:
Better use `qvm-run -u root -a fedora-26-minimal xterm` instead. This method is good for both 3.2 and 4.0.
This will avoid su/sudo/others differences (minimal template in R4.0 do not have passwordless su by default). This do not work with gnome-terminal, unfortunately.